storage/src/main/java/com/google/modernstorage/storage/AndroidFileSystem.kt

```diff
@@ -317,14 +316,13 @@ class AndroidFileSystem(private val context: Context) : FileSystem() {
                 return null
             }
 
-            // DocumentsContract.Document.COLUMN_LAST_MODIFIED
-            val lastModifiedTime = cursor.getLong(0)
-            // DocumentsContract.Document.COLUMN_DISPLAY_NAME
-            val displayName = cursor.getString(1)
-            // DocumentsContract.Document.COLUMN_MIME_TYPE
-            val mimeType = cursor.getString(2)
-            // DocumentsContract.Document.COLUMN_SIZE
-            val size = cursor.getLong(3)
+            val displayName = cursor.getString(cursor.getColumnIndexOrThrow(OpenableColumns.DISPLAY_NAME))
+            val size = cursor.getLong(cursor.getColumnIndexOrThrow(OpenableColumns.SIZE))
+
+            // These two columns are optional and may not be implemented by the providing app.
+            val lastModifiedTime = cursor.getLongOrNull(cursor.getColumnIndex(DocumentsContract.Document.COLUMN_LAST_MODIFIED))
+            val mimeType = cursor.getStringOrNull(cursor.getColumnIndex(DocumentsContract.Document.COLUMN_MIME_TYPE))
+                ?: contentResolver.getType(uri)
```

Comment:
I noticed `ContentResolver#getType(Uri)` is a nice fallback and works in most cases.